python/cudf/cudf/core/buffer.py

```diff
@@ -54,7 +54,7 @@ def __init__(self, data=None, size=None, owner=None):
                 data = memoryview(data)
             except TypeError:
                 raise TypeError("data must be Buffer, array-like or integer")
-            self._init_from_array_like(np.asarray(data), owner)
+            self._init_from_array_like(np.asarray(data).view("|u1"), owner)
```

Comment:
Would suggest not coercing type in the `Buffer` constructor (as we are not doing this for other input types).

```suggestion
            self._init_from_array_like(np.asarray(data), owner)
```